documentation/address-model/resource-definitions.md

```diff
@@ -0,0 +1,65 @@
+# EnMasse resource definition
+
+This document defines the resources consumed and produced by APIs and can be considered a base for writing the schema for the data formats. The [address model](model.md) defines address spaces and addresses which translates to these resources.
+
+The resources defined herein may be created, deleted and updated through different APIs. Either the
+Open Service Broker API, the custom HTTP API, or using AMQP management.
+
+The resources are defined in the JSON format. 
+
+## AddressSpace
+
+The `AddressSpace` resource defines the type of address space and endpoint for connecting.
+
+```
+{
+    "apiVersion": "enmasse.io/v1",
+    "kind": "AddressSpace",
+    "metadata": {
+        "name": "myspace" // Required
+    },
+    "spec": {
+        "type": "standard", // Optional. Default is 'standard'
+        "plan": "small", // Required. This is the aggregate 'size' of the address space (i.e. controls how
+                         // many addresses and how many resources each address may consume
+        "endpoints": { // Optional. If not provided, the default generated hostnames will be provided
```

Comment:
True, we didn't explicitly have them for address-spaces. If these are to be passed from the user then I think it makes sense to add them to the model, and doing so as properties would allow us to evolve without breaking the model.